src/backer/cardaning.py

```diff
@@ -72,43 +70,60 @@ def updateTip(self, tipHeight):
         self.tipHeight = tipHeight
 
     def publishEvent(self, event: bytes):
-        self.pending_kel = event + self.pending_kel
+        self.pending_kel.insert(0, event)
 
-    def submitKelTx(self, kel):
+    def submitKelTx(self):
         try:
-            if kel:
-                # Build transaction
-                builder = pycardano.TransactionBuilder(self.context)
-                builder.add_input_address(self.spending_addr)
-                builder.add_output(pycardano.TransactionOutput(self.spending_addr, pycardano.Value.from_primitive([TRANSACTION_AMOUNT])))
-                kel_data = bytes(kel)
-                # Chunk size
-                # bytearrays is not accept
-                value = [kel_data[i:i + 64] for i in range(0, len(kel_data), 64)]
-
-                # Metadata. accept int key type
-                builder.auxiliary_data = pycardano.AuxiliaryData(pycardano.Metadata({1: value}))
-
-                signed_tx = builder.build_and_sign([self.payment_signing_key],
-                                                change_address=self.spending_addr,
-                                                merge_change=True)
-
-                # Submit transaction
-                self.context.submit_tx(signed_tx.to_cbor())
-                transId = str(signed_tx.transaction_body.inputs[0].transaction_id)
-                trans = {
-                    "id": transId,
-                    "kel": kel.decode('utf-8'),
-                    "tip": self.tipHeight
-                }
-                self.keldbConfirming.pin(keys=transId, val=json.dumps(trans).encode('utf-8'))
+            if self.pending_kel:
+                kel_data = bytearray()
+                submitting_kel = []
+                trans_size = 0
+                tx_cbor = None
+
+                for kel_item in self.pending_kel[:]:
+                    kel_size = len(kel_item)
+
+                    if trans_size + kel_size >= MAX_TRANSACTION_SIZE:
```

Comment:
There's more information in a transaction that just the metadata. You need to construct the full signed transaction and check the size. The signatures also contribute to the size.

Reply:
So keep adding, and once you go over, remove the last one

Reply:
Yes, The signed pycardano tx contains UTXOs, signatures, fee, ...
```
class Transaction(ArrayCBORSerializable):
    transaction_body: TransactionBody

    transaction_witness_set: TransactionWitnessSet

    valid: bool = True

    auxiliary_data: Optional[AuxiliaryData] = None

    @property
    def id(self) -> TransactionId:
        return self.transaction_body.id
```